sstables/sstables.hh

```diff
@@ -625,6 +625,12 @@ private:
     future<> create_data() noexcept;
 
 public:
+    // Finds first position_in_partition in a given partition.
+    // If reversed is false, then the first position is actually the first row.
+    // If reversed is true, then the first position is the last row.
```

Comment:
Note that the static row is always emitted first, even in reverse mode. So this method won't work for partitions with a static row.

Reply:
The fix is simple: when `reversed`, just ignore the static row and use the next fragment's position.

Reply:
I don't think it's a problem. Static row isn't replicated. And the fragment with static row should come before all the others in the ordering.

Reply:
The name of the method, as well as the comments above it suggest otherwise -- that the first/last position is returned. We either have to explain that this method breaks in the presence of static rows or fix it.

Reply:
Static row does have a position, no? So when consuming it, its position is the first one returned. That's what the name of the method suggests.

Reply:
> The fix is simple: when `reversed`, just ignore the static row and use the next fragment's position.

Why when reversed? Ignoring static row should be unconditional.

Reply:
> Static row does have a position, no? So when consuming it, its position is the first one returned. That's what the name of the method suggests.

The method description promises to return the first position when called with `reversed=true` and the last one when `reversed=false`. This will not be the case when there is a static row, as the static row will be both the first and last position (per current implementation).

Reply:
> Why when reversed? Ignoring static row should be unconditional.

Why should it be ignored. Nothing about the name/description suggests that. If that is what is intended you should add `clustering` to the name/description.

Reply:
> > Why when reversed? Ignoring static row should be unconditional.
> 
> Why should it be ignored. Nothing about the name/description suggests that. If that is what is intended you should add `clustering` to the name/description.

That's what you suggested, my friend. "when reversed, just ignore the static row and use the next fragment's position." I am asking about your idea.

Reply:
> > Static row does have a position, no? So when consuming it, its position is the first one returned. That's what the name of the method suggests.
> 
> The method description promises to return the first position when called with `reversed=false` and the last one when `reversed=true`. This will not be the case when there is a static row, as the static row will be both the first and last position (per current implementation).

Agree we need a better description / naming. thanks.

Reply:
I was reacting to the "Ignoring static row should be unconditional" part. It shouldn't be unconditional, we should only ignore it when `reversed=true` and only for the purposes of finding the last position in a partition, which is what this method claims to do.

Reply:
Not sure we should ignore static row. If we accept splitting of partition with static row, then it will be present in a single fragment only that should be opened before all rows. Anyway, will follow with a simpler implementation / better naming and description. thanks

Reply:
And yes, with reversed=true, static row should be ignored. thanks.

Reply:
Please also add a test case with a static row.

Reply:
> I was reacting to the "Ignoring static row should be unconditional" part. It shouldn't be unconditional, we should only ignore it when `reversed=true` and only for the purposes of finding the last position in a partition, which is what this method claims to do.

agree 100%. thanks.

Reply:
> Please also add a test case with a static row.

Will do.